dlrover/python/diagnosis/diagnostician/diagnostician.py

```diff
@@ -0,0 +1,80 @@
+# Copyright 2025 The DLRover Authors. All rights reserved.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import threading
+from abc import ABCMeta, abstractmethod
+from typing import Dict
+
+from dlrover.python.common.log import default_logger as logger
+from dlrover.python.diagnosis.common.diagnosis_action import (
+    DiagnosisAction,
+    NoAction,
+)
+
+
+class Diagnostician(metaclass=ABCMeta):
+    """
+    Diagnostician is to observe problems and resolve those problems
+    """
+
+    def __init__(self):
+        pass
+
+    @abstractmethod
+    def observe(self, **kwargs) -> DiagnosisAction:
+        # observe if particular problem happened
+        return NoAction()
+
+    @abstractmethod
+    def resolve(self, problem: DiagnosisAction, **kwargs) -> DiagnosisAction:
```

Comment:
If the output of the observe method is an action, the resolve method here seems overly redundant(action -> action).

Reply:
Please check the comments above.